src/pages/organize/[orgId]/campaigns/[campId]/callassignments/[callAssId]/callers.tsx

```diff
@@ -0,0 +1,230 @@
+);
+
+interface AssignmentPageProps {
+  assignmentId: string;
+  campId: string;
+  orgId: string;
+}
+
+const TagsCell = ({ tags }: { tags: ZetkinTag[] }) => {
+  const classes = useStyles();
+
+  return (
+    <ZUIResponsiveContainer ssrWidth={200}>
+      {(width) => {
+        const maxTags = Math.floor(width / 100);
+        const displayedTags = tags.slice(0, maxTags);
+        const hiddenTags = tags.slice(maxTags);
+
+        const tooltipTitle = hiddenTags.map((tag) => tag.title).join(', ');
+
+        return (
+          <Box alignItems="center" display="flex" width="100%">
+            {displayedTags.map((tag) => (
+              <TagChip key={tag.id} tag={tag} />
+            ))}
+            {hiddenTags.length > 0 && (
+              <Tooltip title={tooltipTitle}>
+                <Box border={2} className={classes.chip}>
+                  {`${displayedTags.length > 0 ? '+' : ''}${hiddenTags.length}`}
+                </Box>
+              </Tooltip>
+            )}
+          </Box>
+        );
+      }}
+    </ZUIResponsiveContainer>
+  );
+};
+
+const AssignmentPage: PageWithLayout<AssignmentPageProps> = ({
+  orgId,
+  assignmentId,
+}) => {
+  const [onServer, setOnServer] = useState(true);
+  const [searchString, setSearchString] = useState('');
+  const isSearching = searchString.length > 0;
+  const model = useModel(
+    (store) =>
+      new CallAssignmentModel(store, parseInt(orgId), parseInt(assignmentId))
+  );
+  const intl = useIntl();
+
+  useEffect(() => {
+    setOnServer(false);
+  }, []);
+
+  if (onServer) {
+    return null;
+  }
+
+  // Columns
+  const columns: GridColDef[] = [
+    {
+      disableColumnMenu: true,
+      field: 'id',
+      headerName: ' ',
+      renderCell: (params) => (
+        <Avatar src={`/api/orgs/${orgId}/people/${params.id}/avatar`} />
+      ),
+      sortable: false,
+    },
+    {
+      field: 'name',
+      flex: 1,
+      headerName: intl.formatMessage({
+        id: 'pages.organizeCallAssignment.callers.nameColumn',
+      }),
+    },
+    {
+      field: 'prioritizedTags',
+      flex: 1,
+      headerName: intl.formatMessage({
+        id: 'pages.organizeCallAssignment.callers.prioritizedTagsColumn',
+      }),
+      renderCell: (props) => {
+        return <TagsCell tags={props.row.prioritizedTags} />;
+      },
+    },
+    {
+      field: 'excludedTags',
+      flex: 1,
+      headerName: intl.formatMessage({
+        id: 'pages.organizeCallAssignment.callers.excludedTagsColumn',
+      }),
+      renderCell: (props) => {
+        return <TagsCell tags={props.row.excludedTags} />;
+      },
+    },
+  ];
+
+  //Rows
+  const future = model.getFilteredCallers(searchString);
+
+  const rows: GridRowData[] =
+    future.data?.map((caller) => ({
+      excludedTags: caller.excluded_tags,
+      id: caller.id,
+      name: `${caller.first_name} ${caller.last_name}`,
+      prioritizedTags: caller.prioritized_tags,
+    })) ?? [];
+
+  return (
+    <Box>
+      <Paper>
```

Comment:
I think this (and it's contents) could be it's own component, e.g. `CallersSection` or something.